kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/util/NetworkUtil.java

```diff
@@ -101,19 +75,28 @@ public static short getNetmaskShortForm(String netmask) throws KuraException {
     }
 
     public static String dottedQuad(int ip) {
-        StringBuffer sb = new StringBuffer(15);
-        for (int shift = 24; shift > 0; shift -= 8) {
-            // process 3 bytes, from high order byte down.
-            sb.append(Integer.toString(ip >>> shift & 0xff));
-            sb.append('.');
+        String[] items = new String[4];
+        for (int i = 3; i >= 0; i--) {
+            int value = ip & 0xFF;
+            items[i] = Integer.toString(value);
+            ip = ip >>> 8;
         }
-
-        sb.append(Integer.toString(ip & 0xff));
-        return sb.toString();
+        
+        String dottedValue = String.join(".", items);
+        return dottedValue;
     }
 
-    public static int convertIp4Address(String ipAddress) {
+    public static int convertIp4Address(String ipAddress) throws KuraException {
+        if (ipAddress == null) {
+            throw new KuraException(KuraErrorCode.INVALID_PARAMETER);
+        }
```

Comment:
`IllegalArgumentException` is also another possibility. This is also a perfect fit as an unchecked exception. But as a network operations is very low-level work, and the clients of such network operations reside at the high level, it would be good to throw High-level exceptions such as `KuraRuntimeException` or `KuraException` so that higher layers know about such exceptions.

If you believe that the clients of Network Operations are still at the low-level, we can use any relevant standard Java unchecked exceptions such as `IllegalArgumentException` or `NPE` and so on.